test/e2e/storage/volume_metrics.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/metrics"
+)
+
+// This test needs to run in serial because other tests could interfere
+// with metrics being tested here.
+var _ = SIGDescribe("[Serial] Volume metrics", func() {
+	var (
+		c              clientset.Interface
+		ns             string
+		pvc            *v1.PersistentVolumeClaim
+		metricsGrabber *metrics.MetricsGrabber
+	)
+	f := framework.NewDefaultFramework("pv")
+
+	BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+		defaultScName := getDefaultStorageClassName(c)
+		verifyDefaultStorageClass(c, defaultScName, true)
+
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+
+		pvc = newClaim(test, ns, "default")
+		var err error
+		metricsGrabber, err = metrics.NewMetricsGrabber(c, nil, false, false, true, false, false)
+
+		if err != nil {
+			framework.Failf("Error creating metrics grabber : %v", err)
+		}
+	})
+
+	It("should create prometheus metrics for volume provisioning and attach/detach", func() {
+		var err error
+
+		controlerMetrics, err := metricsGrabber.GrabFromControllerManager()
+		if err != nil {
```

Comment:
It's a bit more common to use `Expect` for this type of err  check. Though `Failf` is still seen...